src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java

```diff
@@ -0,0 +1,239 @@
+                    updateResponse.caseDetails,
+                    startEvent
+                );
+
+                return new ProcessResult(
+                    exceptionRecordFinalizer.finalizeExceptionRecord(
+                        existingCase.getData(),
+                        existingCase.getId()
+                    )
+                );
+            }
+        } catch (BadRequest exception) {
+            throw new CallbackException(
+                format(
+                    "Failed to call %s service Case Update API to update case with case Id %s "
+                        + "based on exception record %s",
+                    configItem.getService(),
+                    existingCaseId,
+                    exceptionRecord.id
+                ),
+                exception
+            );
+        } catch (UnprocessableEntity exception) {
+            ClientServiceErrorResponse errorResponse = serviceResponseParser.parseResponseBody(exception);
+            return new ProcessResult(errorResponse.warnings, errorResponse.errors);
+        } catch (Exception exception) {
+            throw new CallbackException(
+                format(
+                    "Failed to update case for %s service with case Id %s based on exception record %s",
+                    configItem.getService(),
+                    existingCaseId,
+                    exceptionRecord.id
+                ),
+                exception
+            );
+        }
+    }
+
+    private void updateCaseInCcd(
+        String service,
+        boolean ignoreWarnings,
+        String idamToken,
+        String s2sToken,
+        String userId,
+        ExceptionRecord exceptionRecord,
+        CaseUpdateDetails caseUpdateDetails,
+        StartEventResponse startEvent
+    ) {
+        CaseDetails existingCase = startEvent.getCaseDetails();
+
+        try {
+            coreCaseDataApi.submitForCaseworker(
+                idamToken,
+                s2sToken,
+                userId,
+                exceptionRecord.poBoxJurisdiction,
+                startEvent.getCaseDetails().getCaseTypeId(),
+                ignoreWarnings,
+                getCaseDataContent(exceptionRecord, caseUpdateDetails, startEvent)
+            );
+
+            log.info(
+                "Successfully updated case for service {} with case Id {} based on exception record ref {}",
+                service,
+                existingCase.getId(),
+                exceptionRecord.id
+            );
+        } catch (FeignException exception) {
+            log.error(
+                "Failed to update case for {} jurisdiction with case Id {} based on exception record with Id {}. "
+                    + "Service response: {}",
+                exceptionRecord.poBoxJurisdiction,
+                existingCase.getId(),
+                exceptionRecord.id,
+                exception.contentUTF8(),
+                exception
+            );
+
+            throw exception;
```

Comment:
Why not throw an exception with the above message, but not have separate logging for it?

Also, we need to handle a 409 differently - that should result with a 200 response from us (see `ProcessResult` class), with an error saying that the case has been modified by someone else.

Reply:
We are catching FeignException here specifically for logging because we can have access to its contentUTF8. I will remove catching Exception below because it is also caught on the level above.

Agree about 409, will do.

Reply:
Why not throw an exception with that message instead?

Reply:
Will do